src/services/rns/rns.processor.ts

```diff
@@ -113,54 +117,55 @@ async function tokenPlacedHandler (logger: Logger, eventData: EventData, eth: Et
   if (!owner) {
     throw new Error(`Domain owner with token ID ${tokenId} not found!`)
   }
+  const offersService = services.offers
+  const currentOffer = await DomainOffer.findOne({ where: { tokenId } })
 
-  const deletedOffersCnt = await DomainOffer.destroy({ where: { tokenId } })
-
-  if (deletedOffersCnt) {
+  if (currentOffer) {
+    await offersService.remove(currentOffer.offerId)
     logger.info(`TokenPlaced event: ${tokenId} previous placement removed`)
   } else {
     logger.info(`TokenPlaced event: ${tokenId} no previous placement`)
   }
+  const { address } = owner
 
-  const domainOffer = new DomainOffer({
+  await offersService.create({
     offerId: transactionHash,
-    sellerAddress: owner.address,
+    ownerAddress: address,
     tokenId: tokenId,
     paymentToken: paymentToken,
     price: cost,
     creationDate: await getBlockDate(eth, eventData.blockNumber)
   })
-  await domainOffer.save()
 
   logger.info(`TokenPlaced event: ${tokenId} created`)
 }
 
-async function tokenUnplacedHandler (logger: Logger, eventData: EventData): Promise<void> {
+async function tokenUnplacedHandler (logger: Logger, eventData: EventData, eth: Eth, services: RnsServices): Promise<void> {
   // event TokenUnplaced(uint256 indexed tokenId);
-
   const tokenId = Utils.numberToHex(eventData.returnValues.tokenId)
-
   const storedOffer = await DomainOffer.findOne({ where: { tokenId } })
 
   if (storedOffer) {
-    storedOffer.destroy()
+    const offersService = services.offers
```

Comment:
You can directly destructuralize the `services` objects into the services you need in the function declaration... I think that would be nicer 😇
(applicable to the whole processor)

Reply:
that was a choice.
it wouldn't be nicer as it would have to be either:
`const { offers } = services`
in which case it's unclear what the constant is, or:
`const { offers: offersService } = services`
which does not even need a comment